packages/@aws-cdk/region-info/build-tools/static-data-facts/promised-region-info.ts

```diff
@@ -0,0 +1,92 @@
+import { IRegionInfo } from '../../lib/iregion-info';
+import { AwsServiceNames } from './index';
+import { servicePrincipal } from './service-principals';
+
+export class PromisedRegionInfo {
```

Comment:
Can you explain why all of these are promises? Are we doing any I/O that this needs to be lazy? Code looks unnecessarily convoluted.

Reply:
Allows me not to have to care about what values depend on what other values in order to be determined. The promise system just resolves everything for me. I can do away with the promises now (because I can just code everything up in a very linear way), but this might make some future additions more awkward to introduce. But I'm cool with the (relatively low) risk here...